modules/cms/classes/Theme.php

```diff
@@ -402,25 +417,63 @@ public function getFormConfig()
         return $config;
     }
 
+    /**
+     * Generates an asset URL for the provided path within the theme, will use the parent theme
+     * if the current theme does not actually have a directory on the filesystem (i.e. is virtual).
+     */
+    public function assetUrl(?string $path): string
+    {
+        $cacheTime = now()->addMinutes(Config::get('cms.urlCacheTtl', 10));
+
+        return Cache::remember('winter.cms.assetUrl.' . $path, $cacheTime, function () use ($path) {
```

Comment:
```suggestion
        $expiresAt = now()->addMinutes(Config::get('cms.urlCacheTtl', 10));
        return Cache::remember('winter.cms.assetUrl.' . $path, $expiresAt, function () use ($path) {
```

Reply:
Also please refactor this to use the same approach as the themeUrl() from Cms\Classes\Controller with the foreach over the sources.